apps/storefront/src/pages/order/Order.tsx

```diff
@@ -94,55 +96,88 @@ function useData() {
     role,
     isAgenting,
     isB2BUser,
-    orderSubViewPermission,
-    selectCompanyHierarchyId: Number(selectCompanyHierarchyId),
-    isEnabledCompanyHierarchy,
-    currentCompanyId: Number(currentCompanyId),
+    isEnabledCompanyHierarchy: isEnabledCompanyHierarchy && orderSubViewPermission,
+    selectedCompanyId: Number(selectCompanyHierarchyId) || currentCompanyId,
     companyId,
   };
 }
 
+const getEmail = (haystack: string = '') => {
+  const getEmailReg = /\((.+)\)/g;
+  const emailRegArr = getEmailReg.exec(haystack);
+
+  return emailRegArr?.length ? emailRegArr[1] : '';
+};
+
+const getName = (haystack: string = '') => {
+  const getCreatedByReg = /^[^(]+/;
+  const createdByUserRegArr = getCreatedByReg.exec(haystack);
+  return createdByUserRegArr?.length ? createdByUserRegArr[0].trim() : '';
+};
+
+interface OrderBy {
+  key: keyof typeof sortKeys;
+  dir: 'asc' | 'desc';
+}
+
+const getOrderBy = ({ key, dir }: OrderBy) => {
+  return dir === 'desc' ? `-${sortKeys[key]}` : sortKeys[key];
+};
+
 function Order({ isCompanyOrder = false }: OrderProps) {
   const b3Lang = useB3Lang();
   const [isMobile] = useMobile();
-  const {
-    role,
-    isAgenting,
-    companyId,
-    isB2BUser,
-    orderSubViewPermission,
-    selectCompanyHierarchyId,
-    isEnabledCompanyHierarchy,
-    currentCompanyId,
-  } = useData();
+  const { role, isAgenting, companyId, isB2BUser, isEnabledCompanyHierarchy, selectedCompanyId } =
+    useData();
+
+  const [pagination, setPagination] = useState<{ offset: number; first: number }>({
+    offset: 0,
+    first: 10,
+  });
 
-  const [isRequestLoading, setIsRequestLoading] = useState(false);
   const [allTotal, setAllTotal] = useState(0);
-  const [filterData, setFilterData] = useState<Partial<FilterSearchProps> | null>(null);
+  const [filterData, setFilterData] = useState<Partial<FilterSearchProps>>();
   const [filterInfo, setFilterInfo] = useState<Array<any>>([]);
   const [getOrderStatuses, setOrderStatuses] = useState<Array<any>>([]);
-  const [isAutoRefresh, setIsAutoRefresh] = useState(false);
 
-  const [handleSetOrderBy, order, orderBy] = useSort(
-    sortKeys,
-    defaultSortKey,
-    filterData,
-    setFilterData,
-  );
+  const [orderBy, setOrderBy] = useState<OrderBy>({
+    key: 'orderId',
+    dir: 'desc',
+  });
+
+  const handleSetOrderBy = (key: string) => {
+    setOrderBy((prev) => {
+      if (!isSortKey(key)) {
+        return prev;
+      }
```

Comment:
is there a use case where this branch would be hit?

If we're just doing it because of the lossy `string` type, we could use an `asserts` instead 🍹 

```tsx
  const handleSetOrderBy = (key: string) => {
    assertIsSortKey(key);
    // etc etc
```